modules/apps/notification/notification-web/src/main/resources/META-INF/resources/js/tests/RequiredFields.spec.ts

```diff
@@ -0,0 +1,150 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2024 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+import '@testing-library/jest-dom/extend-expect';
+
+import {validate} from '../components/EditNotificationTemplate';
+
+const notificationTemplate: NotificationTemplate = {
+	attachmentObjectFieldIds: [],
+	body: {},
+	description: '',
+	editorType: 'richText',
+	externalReferenceCode: '',
+	name: 'name',
+	objectDefinitionExternalReferenceCode: '',
+	objectDefinitionId: null,
+	recipientType: 'email',
+	recipients: [
+		{
+			bcc: '',
+			bccType: 'email',
+			cc: '',
+			ccType: 'email',
+			from: 'user@liferay.com',
+			fromName: {
+				en_US: 'senderName',
+			},
+			singleRecipient: false,
+			to: {
+				en_US: 'user@liferay.com',
+			},
+			toType: 'email',
+		},
+	],
+	subject: {
+		['en_US']: 'subject',
+	},
+	system: false,
+	type: 'email',
+};
+
+describe('Notification template: Validate required fields', () => {
+	it('validates that name field is required', () => {
+		const nameError: NotificationTemplate = {
+			...notificationTemplate,
+			name: '',
+		};
```

Comment:
Hey @igor-franca, I saw you're reppiting a lot this structure, like:

```ts
const nameError = NotificationTemplate = {
			...notificationTemplate,
			name: '',
		};
```

So I'll suggest to you create a function to create tha base of a modified norification template mock, like:

```ts
function generateModifiedNotificationTemplateMock(
	modifiedNotificationTemplateStructure: Partial<NotificationTemplate>
) {
	return {
		...notificationTemplateMock,
		...modifiedNotificationTemplateStructure,
	} as NotificationTemplate;
}
```

and then you can use this way:

obs: note that you are not generating the nameError but the modified NotificationTemplateMock to then get the errors, the validate function will not return just one error, it can return several errors at once, so it would be better to name the errors const from what we receive from the validate to be more consistent with what the code is doing.

```ts
const modifiedNotificationTemplateMock =
	generateModifiedNotificationTemplateMock({
		name: '',
	});
	
const errors = validate(modifiedNotificationTemplateMock);

expect(errors).toStrictEqual({
	name: 'required',
});
```

Reply:
Great suggestions Dan! Thanks